openedx/core/djangoapps/content_libraries/views.py

```diff
@@ -156,14 +158,17 @@ def post(self, request):
         # definitions elsewhere.
         data['library_type'] = data.pop('type')
         data['library_license'] = data.pop('license')
-        # Get the organization short_name out of the "key.org" pseudo-field that the serializer added:
-        org_name = data["key"]["org"]
         # Move "slug" out of the "key.slug" pseudo-field that the serializer added:
         data["slug"] = data.pop("key")["slug"]
+        # Get the organization short_name out of the "key.org" pseudo-field that the serializer added:
+        org_name = data["key"]["org"]
         try:
-            org = Organization.objects.get(short_name=org_name)
-        except Organization.DoesNotExist:
-            raise ValidationError(detail={"org": "No such organization '{}' found.".format(org_name)})
+            ensure_organization(org_name)
```

Comment:
In the other usages of `ensure_organization` earlier in this PR, we get back `org_data` from it.

In this one, we separately get the Organization model based on the org_name after we're sure we haven't thrown an exception... but if we throw an `InvalidOrganizationException`, doesn't that mean the organization doesn't exist?  Is it okay to call `create_library` with a `nil` value?

Reply:
And if it is, can we use `org_data` that came back from `ensure_organization`?  Or is that not an `Organization` model?

Reply:
@davidjoy 

> In this one, we separately get the Organization model based on the org_name after we're sure we haven't thrown an exception... but if we throw an InvalidOrganizationException, doesn't that mean the organization doesn't exist? Is it okay to call create_library with a nil value?

We don't want to create libraries with nil organizations. So when we catch the `InvalidOrganizationException`, we re-raise a `ValidationError` and reject the library creation request.

> And if it is, can we use org_data that came back from ensure_organization? Or is that not an Organization model?

Correct. This bit of code accesses the Organization model directly instead of going through the dictionary-based API, so that it can link to the Organization as a foreign key on the ContentLibrary model. (To color that, everywhere else uses the dict API because edx-organizations has been behind a feature flag, so one couldn't assume the models were available. But the original V2 Content Library authors just went ahead and assumed that the organizations feature was enabled, which is what initiated this whole organization-phase-in shindig).